nether/portal.lua

```diff
@@ -41,6 +41,47 @@ local function save_nether_players()
 	io.close(f)
 end
 
+-- Nether aware mods will need to know if a player is in the nether.
+function nether.is_player_in_nether(player)
+	local pos = player:get_pos()
+	return (pos.y < nether.start) and (pos.y >= nether.bottom)
+end
+
+-- For testing nether trap state tracking.
+function nether.is_player_trapped_in_nether(player)
+	return players_trapped_in_nether[player:get_player_name()]
+end
+
+-- Nether aware mods may have other means of moving players between the Nether
+-- and Overworld, and if so, they should tell us about it so we can keep track
+-- of the player state.
+function nether.external_nether_teleport(player, pos)
+	if not nether.trap_players then
+		player:set_pos(pos)
+		return
+	end
+	local destination_in_nether = (pos.y < nether.start) and (pos.y >= nether.bottom)
+	update_background(player, destination_in_nether)
```

Comment:
`update_background` is out of scope here since the `local update_background` is further down in the code.

>    portal.lua:64:2: accessing undefined variable update_background
>    portal.lua:78:3: accessing undefined variable update_background
>    portal.lua:81:3: accessing undefined variable update_background

Reply:
Okay. I moved that up above the functions that call `update_background()`